### PR TITLE
Parse labels

### PR DESCRIPTION
**Pull request by AndyScherzinger**
Hi @stefan-niedermann, I implemented the parsing of the labels. For some reason they don't find their way to the UI. My guess is I missed something :/

Any hint would be welcome :)

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
JSON from server also does look a bit strange (cc @juliushaertl 
```
{
"title":"Fix some naughty bug",
"description":"getRandom always returns four\nThis is line two\nThis is line three",
"stackId":8,
"type":"plain",
"lastModified":1544169560,
"lastEditor":"andy",
"createdAt":1544169433,
"labels":[
{"title":"To review","color":"317CCC","boardId":"5","cardId":null,"id":18},
{"title":"Later","color":"F1DB50","boardId":"5","cardId":null,"id":20}
],
"assignedUsers":[
{"id":2,"participant":{"primaryKey":"andy","uid":"andy","displayname":"Andy"},"cardId":"24"}
],
"attachments":null,
"attachmentCount":0,
"owner":"andy",
"order":999,
"archived":false,
"duedate":"2018-12-14T07:59:20+00:00",
"deletedAt":0,
"commentsUnread":0,
"id":24,
"overdue":0
}
```

For whatever reason within the label element
* the ```cardId``` of the labels is always `null` and 
* the `boardId` is a `String` instead of a `long`

**Comment by AndyScherzinger** [comment]
As for the JSON/API part I opened an issue for discussion: https://github.com/nextcloud/deck/issues/771

**Comment by stefan-niedermann** [comment]
Hi andy,

thanks for contributing :) displaying labels has already been implemented in https://github.com/stefan-niedermann/nextcloud-deck/pull/17

I think they do not appear in the UI because there are two kind of requests, one to fetch common  information about the cards and one details request. @desperateCoder ?

**Comment by desperateCoder** [comment]
Hi @stefan-niedermann, @AndyScherzinger,
I am currently working on the greenDAO-Branch, where i extended the whole data-model and stuff, I currently can't tell which endpoints you all are talking about. 
Aren't you currently working on mock-data?

**Comment by AndyScherzinger** [comment]
@desperateCoder I added the parser part to the entityparser implementation, see https://github.com/stefan-niedermann/nextcloud-deck/pull/16/files#diff-de7e722624560c9e28e0cf318dbf049bR62 but they don't get passed to the UI (for reasons I don't know)

**Comment by desperateCoder** [comment]
have a look at https://github.com/AndyScherzinger/nextcloud-deck/blob/labels/app/src/main/java/it/niedermann/nextcloud/deck/api/GsonConfig.java

StackArray is mentioned there, but not a single one. Maybe that's it?

Sorry, i would totally checkout your branch and try it myself, but i am currently dealing with a huuuge changeset here, i'd like to get this to a level, so i can merge this into master. This monster is getting out of hand...

**Comment by AndyScherzinger** [comment]
no worries @desperateCoder and thanks for the hint. Unfortunately that doesn't do the trick :/

**Comment by desperateCoder** [comment]
I have a PR running, back to master. As soon as the PR is merged, I will start getting it work. Then you can merge it into yours and I can help you

**Comment by AndyScherzinger** [comment]
Cool, Sounds like a plan. Rebasing the branch after your PR has been merged should be easy.

**Comment by AndyScherzinger** [comment]
Screenshot:
photo_2018-12-09_15-12-35

File: app/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <resources>
 
     <!-- Base application theme. -->
-    <style name="AppTheme" parent="Theme.AppCompat.Light.DarkActionBar">
+    <style name="AppTheme" parent="Theme.MaterialComponents.Light.DarkActionBar.Bridge">
```
Comment (stefan-niedermann) [inline app/src/main/res/values/styles.xml]: Are we getting trouble with our support for older devices? we wanted to support down to minSdk version 14.

Comment (AndyScherzinger) [reply app/src/main/res/values/styles.xml]: Afaik no, Android X is the new AppCompat, so. The Bridge theme is just in place until all. AppCompat widgets have been replaced with the corresponding AndroidX widgets.

Comment (AndyScherzinger) [reply app/src/main/res/values/styles.xml]: @stefan-niedermann for details, see https://developer.android.com/jetpack/androidx/

**Pull request merged**